File: server/DataCore/invops.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models {
	/* invops -  Операции */



	public class invops_register // Регистрация
	{ 
		public System.Guid thePart { get; set; } //Деталь

		[Required]
		public string rfid { get; set; } // RFID детали
		
	}



	public class invops_inventory 
	{

		[Required]
		public Guid InventoryID { get; set; } // инвентаризация

		[Required]
		public string shCode { get; set; } // штрихкод ячейки

		[Required]
		public string rfid { get; set; } // RFID детали

		[Required]
		public int quantity { get; set; } // Количество
	}


	public class invops_clearcell
	{

		[Required]
		public Guid InventoryID { get; set; } // инвентаризация

		[Required]
		public string shCode { get; set; } // штрихкод ячейки

	}


	public class  invops_in { // Приемка
	 public System.Guid  invops_inId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	public string  shCode{ get; set; } // Штрихкод ячейки
	public System.Guid  thePart { get; set; } //Деталь
	
	[Required]
	public string  rfid{ get; set; } // RFID детали
	[Required]
	public int  quantity{ get; set; } // Количество
 }

 public class  invops_move { // Перемещение
	 public System.Guid  invops_moveId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	public string  shCodeFrom{ get; set; } // Штрихкод начальной ячейки 
	[Required]
	public string  shCodeTo{ get; set; } // Штрихкод конечной ячейки
	[Required]
	public string  rfid{ get; set; } // RFID детали

	//[Required]
	//public int  quantity{ get; set; } // Количество
 }

 public class  invops_out { // Отргузка
	 public System.Guid  invops_outId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	public string  shCode{ get; set; } // Штрихкод ячейки
	[Required]
	public string  rfid{ get; set; } // RFID детали
	[Required]
	public int  quantity{ get; set; } // Количество
	public System.Guid  theDept { get; set; } //Отдел
	
 }
}

File: client/src/app/invops.ts
import { enums } from './enums';

export namespace invops { 
	/* invops -  Операции */ 

 export interface   invops_in { // Приемка
	invops_inId:string; // Primary key field
	shCode:string; // Штрихкод ячейки
	thePart:string; //Деталь -> invp_data
	rfid:string; // RFID детали
	quantity:Number; // Количество
	// add dereference fields 
	thePart_name :string; // dereference for invp_data
 }

 export interface   invops_move { // Перемещение
	invops_moveId:string; // Primary key field
	shCodeFrom:string; // Штрихкод начальной ячейки 
	shCodeTo:string; // Штрихкод конечной ячейки
	rfid:string; // RFID детали
	// quantity:Number; // Количество
 }

 export interface   invops_out { // Отгрузка
	invops_outId:string; // Primary key field
	shCode:string; // Штрихкод ячейки
	rfid:string; // RFID детали
	quantity:Number; // Количество
	theDept:string; //Отдел -> invd_dep
	// add dereference fields 
	theDept_name :string; // dereference for invd_dep
 }
}